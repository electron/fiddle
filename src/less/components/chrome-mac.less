.darwin {
  .chrome {
    text-align: center;
    background-color: @background-1;
    border-bottom: @border;
    -webkit-app-region: drag;
    height: 24px;
    font-size: 1.4rem;
    line-height: 15px;

    small,
    p,
    span {
      user-select: none;
    }
  }

  .big_sur_or_later {
    height: 30px !important;
  }
}
