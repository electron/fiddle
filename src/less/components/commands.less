@import "../variables.less";

header {
  box-shadow: 0 0 0 1px rgba(16, 22, 26, 0.2), 0 0 0 rgba(16, 22, 26, 0), 0 1px 1px rgba(16, 22, 26, 0.4);
  background-color: @background-3;
  -webkit-app-region: drag;
}

.disabled-menu-tooltip {
  display: flex;
  width: 100%;

  .bp3-popover-target {
    width: 100%;
  }
}

.commands.is-mac {
  padding:
    10px
    calc(100vw - (env(titlebar-area-x, 0) + env(titlebar-area-width, 0)))
    10px
    env(titlebar-area-x, 0);
}

.commands {
  display: flex;
  justify-content: space-between;
  height: 50px;
  padding:
    10px;
  box-sizing: border-box;
  font-family: @fonts-common;

  button,
  input,
  select {
    -webkit-app-region: no-drag;
    margin-bottom: 0;
    white-space: nowrap;
  }

  select {
    width: 100px;
    margin-left: 5px;
  }

  fieldset {
    border: 0;
    padding: 0;
    margin: 0;
    min-width: 0;
  }

  & > div {
    display: flex;

    &:last-of-type {
      margin-right: 10px;

      .bp3-button-group {
        margin-left: 5px;
      }
    }
  }

  .bp3-control-group {
    margin-right: 0.5rem;
  }

  .address-bar {
    width: 420px;
    transition: width 0.4s;

    input,
    span {
      -webkit-app-region: no-drag;
    }

    &.empty {
      width: 300px;
    }

    &:focus-within {
      span,
      input {
        border-color: @foreground-2;
      }
    }
  }

  & > .title {
    align-self: stretch;
    flex-shrink: 0;

    flex-flow: column nowrap;
    justify-content: center;
    user-select: none;

    margin: 0 10px;
  }
}
