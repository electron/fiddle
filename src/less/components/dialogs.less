@import "../variables.less";

.dialogs {
  z-index: 10;
  position: fixed;
  height: 100vh;
  width: 100vw;
  pointer-events: none;

  * {
    pointer-events: all;
  }
}

.dialog {
  background-color: @background-3;
  border: @border;
  color: @text-color-1;
  border-radius: 6px;
  padding: 10px;
  z-index: 100;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  animation: slide-up 0.4s ease;

  p {
    margin-bottom: 1.25rem;
    font-size: 1.25rem;
  }

  code {
    font-size: 12px;
    background: rgba(255, 255, 255, 0.1);
    padding: 2px;
    line-height: 12px;
  }

  &-buttons {
    margin-top: 1.25rem;
  }

  &.centered {
    position: absolute;
    width: 300px;
    left: calc(50% - (300px / 2));
    top: 100px;
  }

  &.arrow::after,
  &.arrow::before {
    border: solid transparent;
    content: " ";
    height: 0;
    width: 0;
    position: absolute;
    pointer-events: none;
  }

  &.arrow::after {
    border-color: rgba(136, 183, 213, 0);
    border-width: 9px;
  }

  &.arrow::before {
    border-color: rgba(194, 225, 245, 0);
    border-width: 10px;
  }

  &.arrow-top::after,
  &.arrow-top::before {
    bottom: 100%;
    left: 50%;
  }

  &.arrow-top::after {
    border-bottom-color: @background-3;
    margin-left: -9px;
  }

  &.arrow-top::before {
    border-bottom-color: @border-color-1;
    margin-left: -10px;
  }

  &.arrow-bottom::after,
  &.arrow-bottom::before {
    top: 100%;
    left: 50%;
  }

  &.arrow-bottom::after {
    border-top-color: @background-3;
    margin-left: -9px;
  }

  &.arrow-bottom::before {
    border-top-color: @border-color-1;
    margin-left: -10px;
  }

  &.arrow-left::after,
  &.arrow-left::before {
    right: 100%;
    top: 20px;
  }

  &.arrow-left::after {
    border-right-color: @background-3;
    margin-top: -9px;
  }

  &.arrow-left::before {
    border-right-color: @border-color-1;
    margin-top: -10px;
  }

  &.arrow-right::after,
  &.arrow-right::before {
    left: 100%;
    top: 20px;
  }

  &.arrow-right::after {
    border-left-color: @background-3;
    margin-top: -9px;
  }

  &.arrow-right::before {
    border-left-color: @border-color-1;
    margin-top: -10px;
  }

  button {
    color: #1e2527;
    float: right;
    margin-left: 9px;
    margin-bottom: 0;
  }

  a {
    cursor: pointer;
  }

  h4 {
    margin-bottom: 0;
  }

  p:last-of-type {
    margin-bottom: 0;
  }

  input {
    color: rgba(159, 234, 249, 0.5);
    height: 28px;
    margin-top: 10px;
    background-color: @background-1;
    border: 1px solid @border-color-2;
    outline: 0 solid transparent;

    &:focus {
      border-color: @foreground-2;
    }
  }
}

.dialogDrop {
  position: fixed;
  z-index: 100;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: black;
  transition: opacity 0.4s ease-in-out;
  pointer-events: all;
  opacity: 0.5;
}

.dialog-add-version {
  .bp3-file-input {
    width: 100%;
  }

  .bp3-callout {
    margin-top: 1rem;
  }
}

select {
  font-size: 12px;
  height: 28px;
  background-color: white;
  line-height: 15px;
}

@keyframes slide-up {
  0% {
    opacity: 0;
    transform: translateY(20px);
  }

  100% {
    opacity: 1;
    transform: translateY(0);
  }
}
