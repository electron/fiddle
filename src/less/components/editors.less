.mosaic-tile {
  animation: fadein 0.15s;
}

.mosaic-window-toolbar  {
  & > div {
    display: flex;
    justify-content: space-between;
    width: 100%;
    padding: 0 5px 0 5px;

    .mosaic-controls {
      transform: scale(0.75);
      transform-origin: right;
    }

    h5 {
      margin: 4px 0 0 0;
    }

    button {
      margin-left: 8px;
    }
  }
}

// TODO: support new file
// update if list of Editor ID changes
@editor-ids: main\.js, renderer\.js, index\.html, preload\.js, styles\.css;

// takes each editor ID and increase its z-index if the parent Mosaic root
// has focused__id class for that specific id.
each(@editor-ids, {
  .focused__@{value}.mosaic .mosaic-window.@{value} {
    z-index: 2;
  }
});

.mosaic-window {
  z-index: 1;
}

@keyframes fadein {
  from { opacity: 0; }
  to { opacity: 1; }
}
