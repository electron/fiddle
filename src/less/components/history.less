.revision-list {
  max-height: 400px;
  overflow-y: auto;
}

.revision-list ul {
  list-style-type: none;
  padding: 0;
  margin: 0;
}

.revision-item {
  padding: 10px;
  margin-bottom: 5px;
  border-radius: 3px;
  cursor: pointer;
  transition: background-color 0.2s ease;
  border-left: 3px solid #106ba3;
}

.revision-item:hover {
  background-color: rgba(167, 182, 194, 0.3);
}

.revision-content {
  display: flex;
  flex-direction: column;
}

.revision-icon {
  margin-right: 8px;
}

.sha-label {
  font-size: 12px;
  color: #738694;
  margin-left: 10px;
  font-family: monospace;
}

.revision-details {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-top: 5px;
}

.revision-date {
  color: #738694;
  font-size: 12px;
}

.revision-changes {
  display: flex;
  gap: 5px;
}

.history-loading {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;
}
