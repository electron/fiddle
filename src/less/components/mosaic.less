@import '../variables.less';

.fiddle {
  .mosaic-window-toolbar {
    background: var(--background-1);
    box-shadow: inset 0 0 100px 100px rgba(53, 53, 53, 0.1);
    border-top: 1px solid rgba(255, 255, 255, 0.1);
  }

  .mosaic-window-toolbar.draggable:hover {
    box-shadow: inset 0 0 100px 100px rgba(255, 255, 255, 0.1);
  }

  .mosaic-window-title {
    color: var(--text-color-3);
  }

  button.mosaic-default-control {
    padding: 0;
    height: 20px;
    width: 20px;
    margin: 5px;
    background-color: #314246;
    border: 0.1rem solid #222b35;
  }

  button.mosaic-default-control:hover {
    background-color: #131d1f;
  }

  .mosaic-window-body {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    align-items: stretch;
    align-content: stretch;
    overflow: visible;
    z-index: 4;
    background-color: @background-4;
  }

  .mosaic.mosaic-blueprint-theme {
    background: #bdbdbd15;
  }

  .editorContainer,
  .monaco-editor {
    width: 100%;
    height: 100%;
  }

  .editorContainer {
    background: @background-1;
  }
}
