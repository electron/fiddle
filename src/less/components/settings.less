@import "../variables.less";

.settings {
  background: @background-3;
  z-index: 10;
  position: absolute;
  height: 100%;
  width: 100vw;
  display: flex;
  user-select: none;
  cursor: initial;

  a:hover {
    cursor: pointer;
  }

  h1,
  h2,
  h3,
  h4,
  h5 {
    color: @text-color-1;
    cursor: initial;
  }

  h4 {
    margin-bottom: 5px;
  }

  label {
    font-weight: 200;
  }

  select {
    width: 250px;
  }

 .bp4-input-group {
    margin-bottom: 15px;
  }

  .bp4-divider {
    margin-top: 1rem;
    margin-bottom: 1rem;
    margin-left: 0;
    margin-right: 0;
  }

  .bp4-callout {
    max-width: 850px;
  }

  .electron-versions-table {
    width: 100%;

    .action {
      text-align: right;

      .bp4-popover2-target {
        display: block;
      }
    }
  }

  .settings-content {
    height: 100%;
    width: 100%;
    padding: 20px;
    overflow: auto;
    color: @text-color-3;
    box-sizing: border-box;
  }

  .settings-electron {
    .advanced-options-buttons {
      display: flex;
      flex-flow: row wrap;

      > div {
        margin-right: 1rem;

        button {
          width: 100%;
        }
      }

      > div:last-of-type {
        margin-right: 0;
      }
    }

    .disabled-version {
      width: 100%;
    }

    .versions-to-show {
      margin-bottom: 2rem;
      border-bottom: 1px solid @border-color-1;
      padding-bottom: 1.5rem;

      p {
        margin-bottom: 1rem;
      }

      label {
        display: inline;
        margin-right: 10px;
      }
    }
  }

  .settings-section {
    margin-bottom: 20px;

    &:last-of-type {
      margin: 0;
    }
  }

  .settings-close {
    position: fixed;
    right: 30px;
    height: 25px;
    width: 25px;
    text-align: center;
    cursor: pointer;
    border-radius: 15px;
    color: var(--text-color-3);
    padding: 0 0;
    border: 0;

    &:hover {
      color: @text-color-1;
    }

    svg {
      background: @background-3;
      box-shadow: 0 0 12px 6px @background-3;
    }
  }

  .contributors {
    display: flex;
    flex-wrap: wrap;

    .contributor {
      width: 320px;
      height: 120px;
      background: rgba(255, 255, 255, 0.1);
      padding: 10px;
      margin: 0 10px 10px 0;
      display: flex;
      justify-content: flex-start;
      cursor: pointer;

      &:hover {
        background: rgba(255, 255, 255, 0.15);
      }

      h5,
      p {
        margin-bottom: 0;
      }

      .avatar {
        height: 100px;
        width: 100px;
        background-size: contain;
        margin-right: 10px;
      }

      .details {
        max-width: 190px;
        overflow: auto;
      }
    }
  }

  .settings-menu {
    height: 100%;
    width: 250px;
    background-color: @background-1;
    user-select: none;

    ul {
      list-style: none;
      padding: 20px;
    }

    li {
      font-size: 1rem;
      padding-top: 10px;
      padding-bottom: 10px;
      border-bottom: @border;
      color: var(--text-color-1);
      opacity: 0.8;

      &:hover {
        opacity: 1;
      }
    }
  }
}
