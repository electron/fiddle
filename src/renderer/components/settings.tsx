import * as React from 'react';

import { Icon, IconName, MenuItem } from '@blueprintjs/core';
import { observer } from 'mobx-react';

import { AppState } from '../state';
import { CreditsSettings } from './settings-credits';
import { ElectronSettings } from './settings-electron';
import { ExecutionSettings } from './settings-execution';
import { GeneralSettings } from './settings-general';

enum SettingsSections {
  General = 'General',
  Electron = 'Electron',
  Execution = 'Execution',
  Credits = 'Credits',
}

const settingsSections = [
  SettingsSections.General,
  SettingsSections.Electron,
  SettingsSections.Execution,
  SettingsSections.Credits,
];

interface SettingsProps {
  appState: AppState;
}

interface SettingsState {
  section: SettingsSections;
  hasPopoverOpen: boolean;
}

/**
 * Everything above the editors, so buttons and the address bar.
 *
 * @class Settings
 * @extends {React.Component<SettingsProps, SettingsState>}
 */
export const Settings = observer(
  class Settings extends React.Component<SettingsProps, SettingsState> {
    constructor(props: SettingsProps) {
      super(props);

      this.state = {
        section: SettingsSections.General,
        hasPopoverOpen: false,
      };

      this.closeSettingsPanel = this.closeSettingsPanel.bind(this);
    }

    public componentDidMount() {
      window.addEventListener('keyup', this.closeSettingsPanel, true);
    }

    public componentWillUnmount() {
      window.removeEventListener('keyup', this.closeSettingsPanel, true);
    }

    /**
     * Renders the content of the settings, usually by simply
     * return the appropriate component.
     *
     * @returns {(JSX.Element | null)}
     */
    public renderContent(): JSX.Element | null {
      const { section } = this.state;
      const { appState } = this.props;

      if (section === SettingsSections.General) {
        return (
          <GeneralSettings
            appState={appState}
            toggleHasPopoverOpen={() => this.toggleHasPopoverOpen()}
          />
        );
      }

      if (section === SettingsSections.Electron) {
        return <ElectronSettings appState={appState} />;
      }

      if (section === SettingsSections.Execution) {
        return <ExecutionSettings appState={appState} />;
      }

      if (section === SettingsSections.Credits) {
        return <CreditsSettings appState={appState} />;
      }

      return null;
    }

    /**
     * Renders the individual menu items
     *
     * @returns {Array<JSX.Element>}
     */
    public renderOptions(): Array<JSX.Element> {
      const { section } = this.state;

      return settingsSections.map((name) => {
        const isSelected = section === name;
        const onClick = () => this.setState({ section: name });

        return (
          <MenuItem
            onClick={onClick}
            active={isSelected}
            key={name}
            id={`settings-link-${name}`}
            text={name}
            icon={this.getIconForSection(name)}
          />
        );
      });
    }

    public render() {
      const { appState } = this.props;
      const { isSettingsShowing } = appState;

      if (!isSettingsShowing) return null;

      return (
        <div className="settings">
          <div className="settings-menu">
            <ul>{this.renderOptions()}</ul>
          </div>
          <div className="settings-content">
            <button
              className="settings-close"
              onClick={appState.toggleSettings}
            >
              <Icon icon="cross" iconSize={25} />
            </button>
            {this.renderContent()}
          </div>
        </div>
      );
    }

    /**
     * Get the settings icons
     *
     * @param {SettingsSections} section
     * @returns {IconName}
     * @memberof Settings
     */
    private getIconForSection(section: SettingsSections): IconName {
      if (section === SettingsSections.Credits) {
        return 'heart';
      } else if (section === SettingsSections.Electron) {
        return 'floppy-disk';
      } else if (section === SettingsSections.Execution) {
        return 'play';
      }

      return 'cog';
    }

    /**
     * Trigger closing of the settings panel upon Esc
     * If hasPopoverOpen is set to true, settings will not close as only the popover should close
     *
     * @param {KeyboardEvent} event
     */
    private closeSettingsPanel(event: KeyboardEvent) {
      const { appState } = this.props;
      if (event.code === 'Escape' && !this.state.hasPopoverOpen) {
        appState.isSettingsShowing = false;
      }
    }

    /**
     * Toggles whether there is a popover open
     */
    public toggleHasPopoverOpen(): void {
      this.setState({
        ...this.state,
        hasPopoverOpen: !this.state.hasPopoverOpen,
      });
    }
  },
);
