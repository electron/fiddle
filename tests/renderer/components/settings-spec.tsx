import * as React from 'react';

import { shallow } from 'enzyme';

import { Settings } from '../../../src/renderer/components/settings';

jest.mock('../../../src/renderer/components/settings-general', () => ({
  GeneralSettings: 'settings-general',
}));

jest.mock('../../../src/renderer/components/settings-electron', () => ({
  ElectronSettings: 'settings-electron',
}));

jest.mock('../../../src/renderer/components/settings-credits', () => ({
  CreditsSettings: 'settings-credits',
}));

describe('Settings component', () => {
  let store: any;

  beforeEach(() => {
    store = {
      isSettingsShowing: true,
    };
  });

  it('renders null if settings not showing', () => {
    store.isSettingsShowing = false;

    const wrapper = shallow(<Settings appState={store} />);

    expect(wrapper.html()).toBe(null);
  });

  it('renders only the menu if page unknown', () => {
    const wrapper = shallow(<Settings appState={store} />);

    wrapper.setState({ section: 'blub' });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the Electron page by default', () => {
    const wrapper = shallow(<Settings appState={store} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders the General page after a click', () => {
    const wrapper = shallow(<Settings appState={store} />);

    wrapper.find('#settings-link-General').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the Electron page after a click', () => {
    const wrapper = shallow(<Settings appState={store} />);

    wrapper.find('#settings-link-Electron').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the Execution page after a click', () => {
    const wrapper = shallow(<Settings appState={store} />);

    wrapper.find('#settings-link-Execution').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the Credits page after a click', () => {
    const wrapper = shallow(<Settings appState={store} />);

    wrapper.find('#settings-link-Credits').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('closes upon pressing Escape key', () => {
    expect(store.isSettingsShowing).toBe(true);
    // mock event listener API
    const map: any = {};
    window.addEventListener = jest.fn().mockImplementation((event, cb) => {
      map[event] = cb;
    });

    window.removeEventListener = jest.fn().mockImplementation((event) => {
      delete map[event];
    });

    const wrapper = shallow(<Settings appState={store} />);

    // trigger mock 'keyup' event
    map.keyup({ code: 'Escape' });
    expect(Object.keys(map)).toHaveLength(1);
    expect(store.isSettingsShowing).toBe(false);

    // check if event listener is removed upon unmount
    wrapper.unmount();
    expect(Object.keys(map)).toHaveLength(0);
  });

  it('does not close when Escape key is pressed when theme selector is open', () => {
    expect(store.isSettingsShowing).toBe(true);
    // mock event listener API
    const map: any = {};
    window.addEventListener = jest.fn().mockImplementation((event, cb) => {
      map[event] = cb;
    });

    window.removeEventListener = jest.fn().mockImplementation((event) => {
      delete map[event];
    });

    // Set the theme selector showing to true
    const wrapper = shallow(<Settings appState={store} />);
    const instance: any = wrapper.instance();

    // Toggle the state of the variable
    instance.toggleHasPopoverOpen();

    // trigger mock 'keyup' event
    map.keyup({ code: 'Escape' });
    expect(Object.keys(map)).toHaveLength(1);
    expect(store.isSettingsShowing).toBe(true);

    // Toggle the setting again as if it was closed
    instance.toggleHasPopoverOpen();

    // trigger mock 'keyup' event
    map.keyup({ code: 'Escape' });
    expect(Object.keys(map)).toHaveLength(1);
    expect(store.isSettingsShowing).toBe(false);

    // check if event listener is removed upon unmount
    wrapper.unmount();
    expect(Object.keys(map)).toHaveLength(0);
  });
});
