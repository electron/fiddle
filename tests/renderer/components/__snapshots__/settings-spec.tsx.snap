// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders only the menu if page unknown 1`] = `
<div
  className="settings"
>
  <div
    className="settings-menu"
  >
    <ul>
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="cog"
        id="settings-link-General"
        key="General"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="General"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="floppy-disk"
        id="settings-link-Electron"
        key="Electron"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Electron"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="play"
        id="settings-link-Execution"
        key="Execution"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Execution"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="heart"
        id="settings-link-Credits"
        key="Credits"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Credits"
      />
    </ul>
  </div>
  <div
    className="settings-content"
  >
    <button
      className="settings-close"
    >
      <Blueprint4.Icon
        icon="cross"
        iconSize={25}
      />
    </button>
  </div>
</div>
`;

exports[`Settings component renders the Credits page after a click 1`] = `
<div
  className="settings"
>
  <div
    className="settings-menu"
  >
    <ul>
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="cog"
        id="settings-link-General"
        key="General"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="General"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="floppy-disk"
        id="settings-link-Electron"
        key="Electron"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Electron"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="play"
        id="settings-link-Execution"
        key="Execution"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Execution"
      />
      <Blueprint4.MenuItem2
        active={true}
        disabled={false}
        icon="heart"
        id="settings-link-Credits"
        key="Credits"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Credits"
      />
    </ul>
  </div>
  <div
    className="settings-content"
  >
    <button
      className="settings-close"
    >
      <Blueprint4.Icon
        icon="cross"
        iconSize={25}
      />
    </button>
    <settings-credits
      appState={
        Object {
          "isSettingsShowing": true,
        }
      }
    />
  </div>
</div>
`;

exports[`Settings component renders the Electron page after a click 1`] = `
<div
  className="settings"
>
  <div
    className="settings-menu"
  >
    <ul>
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="cog"
        id="settings-link-General"
        key="General"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="General"
      />
      <Blueprint4.MenuItem2
        active={true}
        disabled={false}
        icon="floppy-disk"
        id="settings-link-Electron"
        key="Electron"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Electron"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="play"
        id="settings-link-Execution"
        key="Execution"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Execution"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="heart"
        id="settings-link-Credits"
        key="Credits"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Credits"
      />
    </ul>
  </div>
  <div
    className="settings-content"
  >
    <button
      className="settings-close"
    >
      <Blueprint4.Icon
        icon="cross"
        iconSize={25}
      />
    </button>
    <settings-electron
      appState={
        Object {
          "isSettingsShowing": true,
        }
      }
    />
  </div>
</div>
`;

exports[`Settings component renders the Electron page by default 1`] = `
<div
  className="settings"
>
  <div
    className="settings-menu"
  >
    <ul>
      <Blueprint4.MenuItem2
        active={true}
        disabled={false}
        icon="cog"
        id="settings-link-General"
        key="General"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="General"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="floppy-disk"
        id="settings-link-Electron"
        key="Electron"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Electron"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="play"
        id="settings-link-Execution"
        key="Execution"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Execution"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="heart"
        id="settings-link-Credits"
        key="Credits"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Credits"
      />
    </ul>
  </div>
  <div
    className="settings-content"
  >
    <button
      className="settings-close"
    >
      <Blueprint4.Icon
        icon="cross"
        iconSize={25}
      />
    </button>
    <settings-general
      appState={
        Object {
          "isSettingsShowing": true,
        }
      }
      toggleHasPopoverOpen={[Function]}
    />
  </div>
</div>
`;

exports[`Settings component renders the Execution page after a click 1`] = `
<div
  className="settings"
>
  <div
    className="settings-menu"
  >
    <ul>
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="cog"
        id="settings-link-General"
        key="General"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="General"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="floppy-disk"
        id="settings-link-Electron"
        key="Electron"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Electron"
      />
      <Blueprint4.MenuItem2
        active={true}
        disabled={false}
        icon="play"
        id="settings-link-Execution"
        key="Execution"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Execution"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="heart"
        id="settings-link-Credits"
        key="Credits"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Credits"
      />
    </ul>
  </div>
  <div
    className="settings-content"
  >
    <button
      className="settings-close"
    >
      <Blueprint4.Icon
        icon="cross"
        iconSize={25}
      />
    </button>
    <ExecutionSettings
      appState={
        Object {
          "isSettingsShowing": true,
        }
      }
    />
  </div>
</div>
`;

exports[`Settings component renders the General page after a click 1`] = `
<div
  className="settings"
>
  <div
    className="settings-menu"
  >
    <ul>
      <Blueprint4.MenuItem2
        active={true}
        disabled={false}
        icon="cog"
        id="settings-link-General"
        key="General"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="General"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="floppy-disk"
        id="settings-link-Electron"
        key="Electron"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Electron"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="play"
        id="settings-link-Execution"
        key="Execution"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Execution"
      />
      <Blueprint4.MenuItem2
        active={false}
        disabled={false}
        icon="heart"
        id="settings-link-Credits"
        key="Credits"
        multiline={false}
        onClick={[Function]}
        popoverProps={Object {}}
        shouldDismissPopover={true}
        text="Credits"
      />
    </ul>
  </div>
  <div
    className="settings-content"
  >
    <button
      className="settings-close"
    >
      <Blueprint4.Icon
        icon="cross"
        iconSize={25}
      />
    </button>
    <settings-general
      appState={
        Object {
          "isSettingsShowing": true,
        }
      }
      toggleHasPopoverOpen={[Function]}
    />
  </div>
</div>
`;
